tests/unit_tests/test_client.py

```diff
@@ -135,6 +135,34 @@ def test_uc_is_enabled(unleash_client):
     assert unleash_client.is_enabled("testFlag")
 
 
+@responses.activate
+def test_uc_fallbackfunction(unleash_client, mocker):
+    def good_fallback(feature_name: str, context: dict) -> bool:
+        return True
+
+    def bad_fallback(feature_name: str, context: dict) -> bool:
+        return False
+
+    # Set up API
+    responses.add(responses.POST, URL + REGISTER_URL, json={}, status=202)
+    responses.add(responses.GET, URL + FEATURES_URL, json=MOCK_FEATURE_RESPONSE, status=200)
+    responses.add(responses.POST, URL + METRICS_URL, json={}, status=202)
+    fallback_spy = mocker.Mock(wraps=good_fallback)
+
+    # Create Unleash client and check initial load
+    unleash_client.initialize_client()
+    time.sleep(1)
+    # Only fallback function.
+    assert unleash_client.is_enabled("testFlag", fallback_function=fallback_spy)
+    assert fallback_spy.call_count == 2
+
+    # Default value and fallback function.
+    assert unleash_client.is_enabled("testFlag", default_value=True, fallback_function=bad_fallback)
+
+    # Handle exceptions or invalid feature flags.
+    assert unleash_client.is_enabled("notFoundTestFlag", fallback_function=good_fallback)
+
```

Comment:
would be nice to see at test that name and context is injected. 
Also context should be the enhanced context with static context fields.

Reply:
Test added!